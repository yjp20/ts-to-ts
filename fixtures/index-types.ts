/** @model */
interface Person {
  name: string;
  age: number;
}

/** @model */
type AgeType = Person["age"];

/** @model */
type Properties = keyof Person;

/** @model */
type NameOrAge = Person[keyof Person];
