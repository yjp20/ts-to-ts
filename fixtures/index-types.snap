<<<<<<<<<<< typescript <<<<<<<<<<<<
/** @model */
interface Person {
  name: string;
  age: number;
}

/** @model */
type AgeType = Person["age"];

/** @model */
type Properties = keyof Person;

/** @model */
type NameOrAge = Person[keyof Person];
===================================
// index-types.ts
alias AgeType = float64;

alias Properties = "age" | "name";

alias NameOrAge = string | AgeType;

model Person {
  name: string;
  age: AgeType;
};
>>>>>>>>>>> typespec >>>>>>>>>>>>>>